Formula/libglib.rb

```diff
@@ -22,15 +22,16 @@ class Libglib < Formula
     depends_on "util-linux"
   end
 
-  # replace several hardcoded paths with homebrew counterparts
-  patch do
-    url "https://raw.githubusercontent.com/Homebrew/formula-patches/43467fd8dfc0e8954892ecc08fab131242dca025/glib/hardcoded-paths.diff"
-    sha256 "d81c9e8296ec5b53b4ead6917f174b06026eeb0c671dfffc4965b2271fb6a82c"
-  end
-
   def install
-    inreplace %w[gio/xdgmime/xdgmime.c glib/gutils.c],
-      "@@HOMEBREW_PREFIX@@", HOMEBREW_PREFIX
+    # TODO: This is a workaround for `brew audit --new-formula`.
+    #       Use `patch` rather than `inreplace` (see also `glib`).
+    # replace several hardcoded paths with homebrew counterparts
+    inreplace "gio/xdgmime/xdgmime.c",
+              'xdg_data_dirs = "/usr/local/share/:/usr/share/";',
+              "xdg_data_dirs = \"#{HOMEBREW_PREFIX}/share/:/usr/share/\";"
+    inreplace "glib/gutils.c",
+              'data_dirs = "/usr/local/share/:/usr/share/";',
+              "data_dirs = \"#{HOMEBREW_PREFIX}/share/:/usr/share/\";"
```

Comment:
We shouldn't be working around audits. They exist for a reason. We should send a PR upstream to allow us to configure the setting and use that.

Reply:
I agree, but it may take a much longer time (that's why we allow `patch` blocks rather than send them to every upstream).

The patches given here are copied from `glib.rb` . There is only one match in each corresponding file, so using `inreplace` will provide the same result as `patch`.

Reply:
> I agree, but it may take a much longer time (that's why we allow patch blocks rather than send them to every upstream).

The patches in those blocks should definitely still be submitted upstream. Someone might have skipped that when the `glib` formula was made, but we shouldn't just copy that problem into a new formula.

Reply:
The patch comment used to reference this Bugzilla ticket: https://bugzilla.gnome.org/show_bug.cgi?id=673135